superdesk/core/types/system.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any, Sequence, Protocol
+
+from .web import Request, Endpoint, EndpointGroup
+
+
+class NotificationClientProtocol(Protocol):
+    open: bool
+    messages: Sequence[str]
+
+    def close(self) -> None:
+        ...
+
+    def send(self, message: str) -> None:
+        ...
+
+    def reset(self) -> None:
+        ...
+
+
+class WSGIApp(Protocol):
+    """Protocol for defining functionality from a WSGI application (such as Eve/Flask)
+
+    A class instance that adheres to this protocol is passed into the SuperdeskAsyncApp constructor.
+    This way the SuperdeskAsyncApp does not need to know the underlying WSGI application, just that
+    it provides certain functionality.
+    """
+
+    #: Config for the application
+    config: dict[str, Any]
+
+    #: Config for the front-end application
+    client_config: dict[str, Any]
+
+    testing: bool = False
+
+    #: Interface to upload/download/query media
+    media: Any
+
+    mail: Any
+
+    data: Any
+
+    storage: Any
+
+    auth: Any
+
+    subjects: Any
+
+    notification_client: NotificationClientProtocol
+
+    locators: Any
+
+    celery: Any
+
+    redis: Any
+
+    jinja_loader: Any
+
+    jinja_env: Any
```

Comment:
are we going to provide types for these in the future?

Reply:
Ideally we would not use this at all, by replacing the required functionality into the new async app. For example, `data` will no longer be required as we have equivalent in the new async app.